packages/fiori/src/Timeline.js

```diff
@@ -136,6 +142,48 @@ class Timeline extends UI5Element {
 			}
 		}
 	}
+
+	_onkeyup(event) {
+
+	}
+
+	_onkeydown(event){
+		if (isPageUp(event)) {
+			event.preventDefault();
+			const target = this.items[0];
+			target.focus();
+			this._itemNavigation.setCurrentItem(target);
+		} else if (isPageDown(event)) {
+			event.preventDefault();
+			const target = this.items[this.items.length - 1];
+			target.focus();
+			this._itemNavigation.setCurrentItem(target);
+		} else if (isTabNext(event)) {
+			event.preventDefault()
+			const nextTargetIndex = this.items.indexOf(event.target) + 1
+			const nextTarget = this.items[nextTargetIndex]
+			if (!nextTarget) {
+				return;
+			}
+
+			if (!event.target.nameClickable || event.isMarked === "link") {
+				nextTarget.focus();
+				this._itemNavigation.setCurrentItem(nextTarget)
+			}
+		} else if (isTabPrevious(event)) {
+			event.preventDefault()
+			const nextTargetIndex = this.items.indexOf(event.target) - 1;
```

Comment:
Here it will be previous element, not next. Maybe we can think of a way to extract the code into a function and not have the code repetition.
`
event.preventDefault()
			const nextTargetIndex = this.items.indexOf(event.target) + 1
			const nextTarget = this.items[nextTargetIndex]
			if (!nextTarget) {
				return;
			}
			if (!event.target.nameClickable || event.isMarked === "link") {
				nextTarget.focus();
				this._itemNavigation.setCurrentItem(nextTarget)
			}
`
It's either + or - 1 so it can be done with a boolean variable, and calculated on the run. Also i see that there's a missing check for the 'nameClickable' for the previous one, and i think we also should check for the title, even if we go back? The function can be called something like "moveToNextElement(target, direction)"